server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -552,23 +560,88 @@ public boolean equals(Object obj) {
                 return false;
             }
             TimeIntervalRounding other = (TimeIntervalRounding) obj;
-            return Objects.equals(interval, other.interval) && Objects.equals(timeZone, other.timeZone);
+            return interval == other.interval && timeZone.equals(other.timeZone);
+        }
+
+        @Override
+        public String toString() {
+            return "Rounding[" + interval + " in " + timeZone + "]";
```

Comment:
The `toString` implementations weren't consistent which made reading the error messages hard. So I changed them.